src/app/highchart/highchart.component.ts

```diff
@@ -86,22 +91,40 @@ export class HighchartComponent implements OnInit {
           };
           s = s + Highcharts.dateFormat('%Y', this.x) + '';
           this.points.forEach((point) => {
-            let label = '<br>' + point.series.name + ': ' + Highcharts.numberFormat(point.y);
+            let name;
+            if (point.series.name === 'Level') {
+              name = '';
+            }
+            if (point.series.name === 'YTD' && dataFreq === 'A') {
+              name = 'Year-over-Year';
+            }
+            if (point.series.name === 'YTD' && dataFreq !== 'A') {
+              name = 'Year-to-Date';
+            }
+            if (point.series.name === 'YTD' && percent) {
+              name += ' Chg: ';
+            }
+            if (point.series.name === 'YTD' && !percent) {
+              name += ' % Chg: ';
+            }
```

Comment:
The above section could be replaced by a function with early returns. The if statements are mutually exclusive so at a minimum using else if would prevent each case from being tested. Using a function with early returns is closer to the popular best practice.